api/registrations/serializers.py

```diff
@@ -397,15 +397,19 @@ class RegistrationSerializer(BaseRegistrationSerializer):
     registration_choice = ser.ChoiceField(write_only=True, choices=['immediate', 'embargo'])
     lift_embargo = VersionedDateTimeField(write_only=True, default=None, input_formats=['%Y-%m-%dT%H:%M:%S'])
 
-    # Because the write-only `children` field shadows the read-only both fields must be included here to prevent the
-    # write-only from overriding the read-only in the parent class.
-    children = HideIfWithdrawal(RelationshipField(  # Read-only
-        related_view='registrations:registration-children',
-        related_view_kwargs={'node_id': '<_id>'},
-        related_meta={'count': 'get_node_count'},
-    ))
-
-    children = ser.ListField(write_only=True, required=False)  # Write-only
+    def __init__(self, *args, **kwargs):
```

Comment:
Instead of overriding `__init__`, how about making a separate `RegistrationCreateSerializer` to handle the writeable children field? We do this for node contributors (see [here](https://github.com/CenterForOpenScience/osf.io/blob/develop/api/nodes/serializers.py#L1055) and [here](https://github.com/CenterForOpenScience/osf.io/blob/develop/api/nodes/views.py#L415)) and this way it's better separated in case the serializer needs to deviate further based on the request method.